Tests/FluentAssertions.Specs/Exceptions/AsyncFunctionExceptionAssertionSpecs.cs

```diff
@@ -1187,6 +1187,48 @@ public async Task When_no_exception_should_be_thrown_on_UI_thread_for_async_func
             // Assert
             await act.Should().NotThrowAsync();
         }
+
+        [Fact]
+        public async Task When_async_method_throws_the_exact_expected_inner_exception_it_should_succeed()
+        {
+            // Arrange
+            Func<Task> task = async () =>
+            {
+                await Task.Delay(100);
```

Comment:
I'm unsure why you've added a delay here.

I believe this would do.
```cs
Func<Task> task = () => FromException(new AggregateException(new InvalidOperationException()));
```

Reply:
Just followed the test structure that were already present. Fixed.

Reply:
Oh I see, my apologies then.

@dennisdoomen Do you by any change recall why the 100ms delay was inserted in the first place in #515?
I see that [`Task.Exception`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.fromexception?view=netframework-4.6) was first introduced in .NET 4.6 and [`Task.Yield()`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.yield?view=netframework-4.6) in .NET 4.5, where #515 was released with FA 4.18 where we supported .NET 4.0.
But since its 100ms and not just 0ms I'm curious if there is some technical detail I'm not seeing.

Reply:
I'm not 100% sure, but I suspect I did this to ensure the `async` function block wasn't executed synchronously.